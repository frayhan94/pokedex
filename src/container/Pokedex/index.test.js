import React from 'react';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Poke from './index';

const mockStore = configureMockStore();

describe('RENDER POKE DASHBOARD', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    const initialState = {
      pokedexReducer: {
        loadingPokedex: false,
        pokedex: [],
        totalPokedex: 0,
        offset: 0,
        limit: 100,
      },
    };
    store = mockStore(initialState);
    store.dispatch = jest.fn();
    wrapper = renderer.create(
      <Provider store={store}>
        <Poke />
      </Provider>,
    );
  });

  it('RENDER CORRECTLY', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
