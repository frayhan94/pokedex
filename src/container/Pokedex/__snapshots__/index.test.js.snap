// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDER POKE DASHBOARD RENDER CORRECTLY 1`] = `
Array [
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Pokedex
  </h3>,
  <div
    className="Grid justifyCenter split--5"
  />,
]
`;
